Docker/run_supervisord.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+if [[ "$PGBOUNCER_ENABLE" == "true" || "$PGBOUNCER_ENABLE" == "TRUE" ]]; then
+    echo "PGBOUNCER_ENABLE env is true, using /etc/supervisor/conf.d/adaguc-pgbouncer.conf"
+
+    # Look up db host/port/user/password/name from ADAGUC_DB string
+    read -r db_host db_port db_user db_password db_name <<< $(echo $ADAGUC_DB | awk -F'[= ]' '{ print $2" "$4" "$6" "$8" "$10 }')
+
+    # Disable ssl for pgbouncer when running through docker-compose
+    # if host=adaguc-db, then we're using docker-compose.
+    if [[ "$db_host" == "adaguc-db" ]]; then
+        sed -i "s/^server_tls_sslmode = .*/server_tls_sslmode = disable/g" /adaguc/pgbouncer/pgbouncer.ini
+    fi
+
+    # Set username/password, pgbouncer uses this to authenticate to RDS
+    echo '"'$db_user'"' '"'$db_password'"' > /adaguc/pgbouncer/userlist.txt
+
+    # Rewrite pgbouncer config. $db_host could be an RDS hostname or adaguc-db when running through docker-compose.
+    pgbouncer_cfg="adaguc = host=$db_host port=$db_port dbname=$db_name"
+    sed -i "s/^adaguc = .*/${pgbouncer_cfg}/g" /adaguc/pgbouncer/pgbouncer.ini
+
+    # Rewrite $ADAGUC_DB
+    export ADAGUC_DB="host=localhost port=${db_port} user=${db_user} password=${db_password} dbname=${db_name}"
```

Comment:
This part seems a little bit nasty to me.

`ADAGUC_DB` gets set through `docker compose` or the task definition when running in AWS. I extract the `host=...` section to instruct pgbouncer.
I then rewrite `ADAGUC_DB` to just connect to `host=localhost`.

It works for adaguc-server, it connects to localhost. However if you login to the container `echo $ADAGUC_DB`, it will still show `host=rds...` and not the rewritten variable.

I can't find a way to "rewrite" this environment variable and make docker remember it. Maybe it's not an issue but good to keep in mind.

Reply:
I don't think it is possible to change environment of the parent shell, so I guess this can not be avoided.